packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.menu.tests.js

```diff
@@ -595,6 +599,85 @@ QUnit.module('widget sizing render', moduleConfig, () => {
 
         });
 
+        QUnit.test('space on list item hides drop-down list (T1301705)', function(assert) {
+            if(shouldSkipOnMobile(assert)) {
+                return;
+            }
+
+            this.instance.option('closeOnClick', true);
+            this.instance.option('opened', false);
+            this.overflowMenu.$button().focusin();
+
+            this.keyboard.keyDown('enter');
+            const popup = this.overflowMenu.popup();
+            assert.equal(popup.option('visible'), true, 'popup is visible');
+
+            this.instance.option(
+                'items', [
+                    {
+                        template: () => $('<div>').dxCheckBox({ value: false }),
+                        widget: 'dxCheckBox',
+                    },
+                    {
+                        template: () => $('<div>').dxButton({ text: 'Button' })
+                    },
+                ]
+            );
+
+            const checkBox = this.overflowMenu.$popupContent().find(`.${CHECK_BOX_CLASS}`).first();
+            const button = this.overflowMenu.$popupContent().find(`.${BUTTON_CLASS}`).first();
+
+            this.instance.option({ focusStateEnabled: true });
+            this.overflowMenu.list().option({ focusStateEnabled: true });
+
+            // checkBox.focusin();
+            keyboardMock(checkBox)
+                .keyDown('space');
+            assert.equal(popup.option('visible'), true, 'space on dxCheckbox does not hide popup');
```

Comment:
Please let's try to keep tests as atomic as possible. For this the following practices can help:
1. AAA pattern (arrange - act - assert)
2. describing the whole test in the name. In your case the correct name for this test is smth like "popup should not be closed after space key press on a list item nested checkbox but should be closed after press on a list item nested button"